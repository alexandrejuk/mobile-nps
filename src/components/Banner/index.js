import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import BannerSvg from '../../assets/images/banners/banner-1.svg'
import styles from './style.module.css'

const Banner = ({
  fill,
}) => (
  <div 
    className={classNames(
      styles.banner,
      styles[fill],
    )}
  >
    <img 
      src={BannerSvg} 
      alt="banner information" 
    />
  </div>
)

Banner.propTypes = {
  fill: PropTypes.oneOf([
    'primary',
    'secondary',
  ]).isRequired,
}

export default Banner 
