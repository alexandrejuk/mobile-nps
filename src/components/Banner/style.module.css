@import '../../assets/css/colors.css';

.banner {
  background-color: var(--color-white);
  border-radius: 8px;
  box-sizing: border-box;
  box-shadow: 0px 2.27027px 2.27027px rgba(0, 0, 0, 0.24), 0px 0px 2.27027px rgba(0, 0, 0, 0.12);
  height: 159px;
  text-align: center;
  overflow: hidden;
  width: 100%;
}

.primary {
  background-color: var(--color-california);
}

.secondary {
  background-color: var(--color-cardinal);
}

.banner img {
  height: 100%;
}
