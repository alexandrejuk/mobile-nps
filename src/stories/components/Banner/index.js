import React from 'react'
import { Banner } from '../../../components'

export default {
  title: 'Components/Banner',
}

export const Primary = () => (
  <Banner fill="primary" />
)
