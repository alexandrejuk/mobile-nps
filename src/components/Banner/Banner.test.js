import React from 'react'
import { render } from '@testing-library/react'

import Banner from './'

describe('Banner component', () => {
  it('should render correctly primary banner', () => {
    const { container } = render(
      <Banner fill="primary" />
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly secondary banner', () => {
    const { container } = render(
      <Banner fill="secondary" />
    )
    expect(container).toMatchSnapshot()
  })
})
