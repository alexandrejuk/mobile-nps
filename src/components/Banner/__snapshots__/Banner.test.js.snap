// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component should render correctly primary banner 1`] = `
<div>
  <div
    class="banner primary"
  >
    <img
      alt="banner information"
      src="banner-1.svg"
    />
  </div>
</div>
`;

exports[`Banner component should render correctly secondary banner 1`] = `
<div>
  <div
    class="banner secondary"
  >
    <img
      alt="banner information"
      src="banner-1.svg"
    />
  </div>
</div>
`;
